packages/slice-machine/server/src/api/common/generate.ts

```diff
@@ -20,14 +22,21 @@ export function generate(
     );
 
     components.forEach((c) => {
-      if (
-        !Files.exists(
-          // eslint-disable-next-line @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
-          GeneratedPaths(env.cwd).library(c.from).slice(c.model.name).mocks()
-        )
-      ) {
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-        const mocks = mock(
+      const mocksPath = CustomPaths(env.cwd)
+        .library(c.from)
+        .slice(c.model.name)
+        .mocks();
+      const mocks = Files.readEntityFromFile<ComponentMocks>(
+        mocksPath,
+        (payload: unknown) => {
+          return getOrElseW(() => new Error("Invalid component mocks."))(
+            ComponentMocks.decode(payload)
+          );
+        }
+      );
+
+      if (!mocks || mocks instanceof Error) {
+        const mocks: ComponentMocks = mock(
```

Comment:
is there no error for mocks being declared twice?

Reply:
oh yeah weird. I'll do some renaming around here.